### ISSUE TITLE
Build failure in store-0.7.2 with GHC 8.10.1

### ISSUE THREAD

**Issue by sjakobi** [mgsloan/store#149]
```
src/Data/Store/Internal.hs:851:3: error:
    • Expecting one more argument to ‘IO’
      Expected a type, but ‘IO’ has kind ‘* -> *’
    • In the first argument of ‘Store’, namely ‘IO’
      In the instance declaration for ‘Store IO’
    |
851 | $(reifyManyWithoutInstances ''Store [''Info] (const True) >>=
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/Data/Store/Internal.hs:851:3: error:
    • Expecting two more arguments to ‘GHC.STRef.STRef’
      Expected a type, but ‘GHC.STRef.STRef’ has kind ‘* -> * -> *’
    • In the first argument of ‘Store’, namely ‘GHC.STRef.STRef’
      In the instance declaration for ‘Store GHC.STRef.STRef’
    |
851 | $(reifyManyWithoutInstances ''Store [''Info] (const True) >>=
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/Data/Store/Internal.hs:851:3: error:
    • Expecting one more argument to ‘GHC.IORef.IORef’
      Expected a type, but ‘GHC.IORef.IORef’ has kind ‘* -> *’
    • In the first argument of ‘Store’, namely ‘GHC.IORef.IORef’
      In the instance declaration for ‘Store GHC.IORef.IORef’
    |
851 | $(reifyManyWithoutInstances ''Store [''Info] (const True) >>=
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

src/Data/Store/Internal.hs:851:3: error:
    • Expecting one more argument to ‘GHC.ForeignPtr.ForeignPtr’
      Expected a type, but ‘GHC.ForeignPtr.ForeignPtr’ has kind ‘* -> *’
    • In the first argument of ‘Store’, namely
        ‘GHC.ForeignPtr.ForeignPtr’
      In the instance declaration for ‘Store GHC.ForeignPtr.ForeignPtr’
    |
851 | $(reifyManyWithoutInstances ''Store [''Info] (const True) >>=
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

**Comment by RyanGlScott** [comment]
This is most likely due to `template-haskell-2.16.0.0` introducing the [`Bytes`](https://downloads.haskell.org/~ghc/8.10.1/docs/html/libraries/template-haskell-2.16.0.0/Language-Haskell-TH-Syntax.html#t:Bytes) data type to `Lit`, whose constructor uses `ForeignPtr`. Since there is no `Store` instance for `Bytes`, `reifyManyWithoutInstances` attempts to recursively define `Store` instances for `Bytes` and everything mentioned in its constructor. This includes `ForeignPtr`, `IO`, and friends.

**Comment by mgsloan** [comment]
Yep, that's exactly what's happening @RyanGlScott - I'm working on fixing it.

**Comment by mgsloan** [comment]
I've written the instances, the thing I'm encountering right now is that `Bytes` uses derived instances of Eq / Ord / etc which are comparing pointers and such rather than the contents of the bytes.  Which sucks, because it breaks the roundtrip properties.

**Comment by mgsloan** [comment]
Fixed in `store-0.7.3`!

**Comment by mgsloan** [comment]
Oops, didn't use enough CPP.  This time I will wait until travis is green before pushing to hackage

**Comment by mgsloan** [comment]
Properly fixed in `store-0.7.4`

**Comment by sjakobi** [comment]
Thank you! :)
